### PR TITLE
Manual control 24

### PR DESCRIPTION
**Pull request by vashmata**

### CODE REVIEW

File: .vscode/c_cpp_properties.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
```
Comment (PeterGhimself) [inline .vscode/c_cpp_properties.json]: I think this should not be included, but would have to double check with @tatumalenko

File: .vscode/settings.json [hunk 2]
```diff
@@ -6,8 +6,16 @@
     "python.linting.lintOnSave": true,
     "python.linting.enabled": true,
     "python.linting.pylintEnabled": true,
+<<<<<<< HEAD
```
Comment (PeterGhimself) [inline .vscode/settings.json]: This needs resolving

File: README.md [hunk 3]
```diff
@@ -2,9 +2,18 @@
 This repo contains the Robotics software team code, as well as some of the other subteams code/documentation.
 
 ## Contributing and Development Environment Instructions
-Firstly, this project is built in Python 3.6+. You need to have a version of Python installed that is 3.6+. Make sure that whenever you use `python` or `python3` or whatever later on meets this requirement.
 
-Secondly, it is imperative you use a virtual env (instead of your system Python) to use/contribute to the project. 
+Firstly, this project is built in Python 3.3+ and JavaScript (ES6). You need to have a version of Python installed that is 3.6+ and Node + NPM (see [here](https://nodejs.org/en/download/)). Make sure that whenever you use `python` or `python3` or whatever later on meets this requirement.
```
Comment (PeterGhimself) [inline README.md]: This line should be "built in Python 3.6+"

File: robot/archives/gtk/Motor.py [hunk 4]
```diff
@@ -7,6 +7,12 @@
 # working software over comprehensive documentation
 # customer collaboration over contract negotiation
 # responding to change over following a plan
+import struct
+
+'''
+This class represents a motor component that will be used inside
+the mother `Microcontroller` class container (such as the Arduino/ODROID).
```
Comment (PeterGhimself) [inline robot/archives/gtk/Motor.py]: Though the odroid is not an MCU

File: robot/demos/RemoteArmControl/SingleFile.py [hunk 5]
```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python3
```
Comment (PeterGhimself) [inline robot/demos/RemoteArmControl/SingleFile.py]: Not sure what the point is here (or rather if it was achieved, based off the name of the file)

File: robot/demos/RemoteArmControl/comms.sh [hunk 6]
```diff
@@ -0,0 +1,2 @@
+terminator -T "ServerListenerWindow" -e "./ServerListenerLocal.py" &
```
Comment (PeterGhimself) [inline robot/demos/RemoteArmControl/comms.sh]: This script should start with a shebang.
Adding `#!/usr/bin/env bash` as the first line will do the trick.

File: .vscode/c_cpp_properties.json (hunk 1 above)
Comment (tatumalenko) [reply .vscode/c_cpp_properties.json]: Yah I wouldn't either lol

File: robot/demos/RemoteArmControl/SingleFile.py (hunk 5 above)
Comment (vashmata) [reply robot/demos/RemoteArmControl/SingleFile.py]: I was making a script that doesn't have any udp comms between 2 scripts in order to test things out.

File: robot/rover/ArmDriverUnit/Examples/RosCommunication/RosCommunication.ino [hunk 7]
```diff
@@ -0,0 +1,307 @@
...
+
+// I define this here because that's how the examples do it.
+// I think it has to be before the subscriber is instantiated but usually i put function definitions down below. test leaving it as just a declaration?
+// I think this is called when you use nh.spinOnce() and a message with the appropriate topic was received
+void messageCallback(const std_msgs::String& cmd_message) {
+  msgReceived = true;
+  int i = 0;
+  // place the message into serialBuffer for parsing in setup()
+  while (cmd_message.data[i] != '\0') {
+    serialBuffer[i] = cmd_message.data[i];
```
Comment (PeterGhimself) [inline robot/rover/ArmDriverUnit/Examples/RosCommunication/RosCommunication.ino]: I'm assuming you meant to but forgot to increment `i` here?

File: robot/rover/ArmDriverUnit/Examples/ServoExampleOpenLoop/PidController.h [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+  jointAngleTolerance = 1.0;
+  minOutputValue = -50.0;
+  maxOutputValue = 50.0;
+  slowestSpeed = 3.0;
+}
+
+float PidController::updatePID(volatile float currentAngle, float desiredAngle)
+{
+  float pidOutput;
+  error = desiredAngle - currentAngle; // these angle variables need to be obtained from the notor object
```
Comment (PeterGhimself) [inline robot/rover/ArmDriverUnit/Examples/ServoExampleOpenLoop/PidController.h]: the notor object, myes indeed

**Pull request merged**